modules/sql-engine/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -675,19 +675,32 @@ SqlTypeNameSpec IgniteDateTimeTypeName() :
 |
     <TIME> { s = span(); }
     precision = PrecisionOpt()
+    (<WITHOUT> <TIME> <ZONE>)?
     {
         typeName = SqlTypeName.TIME;
         return new SqlBasicTypeNameSpec(typeName, precision, s.end(this));
     }
 |
     <TIMESTAMP> { s = span(); }
     precision = PrecisionOpt()
-    typeName = TimeZoneOpt(false)
+    typeName = TimestampZoneOpt()
     {
         return new SqlBasicTypeNameSpec(typeName, precision, s.end(this));
     }
 }
 
+SqlTypeName TimestampZoneOpt() :
+{
+}
+{
+    LOOKAHEAD(3)
```

Comment:
why do we need it ? did i miss smth ?

Reply:
Thanks, looks redundant.
Was copy pasted from calcite :roll_eyes: